raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -343,16 +347,14 @@ private void handleVoteResponse(int remoteNodeId, VoteResponseData response) thr
         } else {
             CandidateState state = quorum.candidateStateOrThrow();
             if (response.voteGranted()) {
-                state.voteGrantedBy(remoteNodeId);
+                state.recordGrantedVote(remoteNodeId);
                 maybeBecomeLeader(state);
             } else {
-                state.voteRejectedBy(remoteNodeId);
+                state.recordRejectedVote(remoteNodeId);
                 if (state.isVoteRejected()) {
-                    logger.info("A majority of voters rejected our candidacy, so we will become a follower");
-                    becomeUnattachedFollower(quorum.epoch());
-
-                    // We will retry our candidacy after a short backoff if no leader is elected
-                    electionTimer.reset(retryBackoffMs + randomElectionJitterMs());
+                    logger.info("Insufficient remaining votes to become leader (rejected by {}). " +
```

Comment:
Note that I removed the logic to reset the timer here. Even if our candidacy has been rejected, we cannot conclude that another leader cannot emerge, so we do not want to be too aggressive about starting a new election. Instead we just await the election timeout like normal.